### DISCUSSION TITLE
How can I sample continuous transitions

### DISCUSSION THREAD

**Discussion by heyuanYao-pku** [ymd-h/cpprb#21] [Q&A] [answered]
I'm writing a custom training process, which is autoregressive . And it needs to sample $n$ continuous transitions from the replay buffer so that it can be used as the ground truth of roll out.
 $n$ is perhaps 16 or 24 and so on.....

**Comment by heyuanYao-pku** [comment]
Or perhaps, is there a convenient way to sample from buffer with a custom function?

**Comment by ymd-h** [comment, marked as answer]
@heyuanYao-pku 
Hi, thank you for your feed backs.

I don't find any convenient ways, but there are workarounds.

---

Option 1:
You can transform to $n$-length batch shape before adding.
Additional local buffer can be used, however, if you want to utilize `stack_compress` option, stack dimension must be the last dimension.
https://ymd_h.gitlab.io/cpprb/features/memory_compression/

---

Option 2:
You can use private method `_encode_sample`, which takes index.
You might generate random number outside, or slide PER indices after sampling.

Note:
`_encode_sample` is private method and not guaranteed within API compatibility, however, I know some users use the method, so that we don't change without enough reason.

---

FYI:
(I am not sure about the detail of your needs, so it might be nonsense for you.)

For online periodical roll out, some users utilize `ReplayBuffer`.
When buffer becomes full, dump with `get_all_transitions()` method, then `clear()` it.
